test: Reduce number of C++ test cases

Funatiq opened the pull request:
## Description

- Always enable TRT overlap and chunked context in `trtGptModelRealDecoderTest`.
- Use only `InflightFusedBatching` in most `trtGptModelRealDecoderTest` tests. 
- Improve documentation in executor tests.
- Refactor `TokenComparison` executor test and improve output verification (Fix log probs verification with `returnAllGeneratedTokens` enabled).
-  Always enable `computeLogProbs`, `returnContextLogits`, and `returnGenerationLogits` in executor tests.
- Re-enable executor leader tests.
- Refactor ParamCancelReqTest executor test (Create requests with different parameters instead of using test parameters).
- Refactor LeaderApiUsageTest and enable (Create requests with different parameters instead of using test parameters).    

## Test Durations 

Before numbers from post-merge 2127.

| Test Case | Before | After |
|-----------|--|--|
| A30-CPP-1/cpp/test_e2e.py::test_model[-gpt_executor-80] | 2596.86s | 1556.37s |
| A30-CPP-2/cpp/test_e2e.py::test_model[-gpt-80] | 2196.60s | 1503.43s |
| A30-CPP-3/cpp/test_e2e.py::test_model[-gpt_tests-80] | 1595.81s | 1226.72s |
| H100_PCIe-CPP-2/cpp/test_e2e.py::test_model[fp8-llama-90] | 944.47s | 384.55s |
| DGX_H100-4_GPUs-CPP-1/cpp/test_multi_gpu.py::test_trt_gpt_real_decoder[llama-90] | 327.91s | 167.66s |
| DGX_H100-4_GPUs-CPP-1/cpp/test_multi_gpu.py::test_llama_executor[llama-orchestrator-90] | 526.56s | 296.48s |
| DGX_H100-4_GPUs-CPP-1/cpp/test_multi_gpu.py::test_llama_executor[llama-leader-90] | - | 125.93s |

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

Funatiq committed:
test: Reduced the number of test cases in trtGptModelRealDecoderTest

- Always enable TRT overlap and chunked context options for various test cases.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
test: Improve documentation in executor tests

- Added detailed comments for each parameter type to enhance understanding of their purpose.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
test: Reduced number of test cases in executor tests

- Updated test instantiation to always enable `computeLogProbs`, `returnContextLogits`, and `returnGenerationLogits` for both Gpt and Llama executor tests.
- Removed redundant skips for `returnAllGeneratedTokens` with `outConfig.returnLogProbs`.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #9693](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9693/) [ run ] triggered by Bot

Funatiq committed:
test: Improve executor test output verification

- Simplified the handling of request IDs and batch IDs in the `runTest` function.
- Updated the `verifyOutput` method to remove redundant checks for `returnAllGeneratedTokens`.
- Fix log probs verification with `returnAllGeneratedTokens` enabled.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run --stage-list "A10-CPP-1, A30-CPP-1, A30-CPP-2, A30-CPP-3, H100_PCIe-CPP-1, H100_PCIe-CPP-2, DGX_H100-4_GPUs-CPP-1"

tensorrt-cicd commented:
[PR_Github #9718](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9718/) [ run ] triggered by Bot

Funatiq commented:
/bot run --stage-list "A10-CPP-1, A30-CPP-1, A30-CPP-2, A30-CPP-3, H100_PCIe-CPP-1, H100_PCIe-CPP-2, DGX_H100-4_GPUs-CPP-1"

tensorrt-cicd commented:
[PR_Github #9730](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9730/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #9875](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9875/) [ run ] triggered by Bot

Funatiq committed:
fixup! test: Reduced number of test cases in executor tests

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
test: Re-enable executor leader tests

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run --stage-list "DGX_H100-4_GPUs-CPP-1"

DomBrown reviewed (approved):
LGTM, I assume you're happy with the level of coverage.

tensorrt-cicd commented:
[PR_Github #9892](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9892/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7300 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7300/) completed with status: 'FAILURE'

Funatiq committed:
test: Refactor ParamCancelReqTest executor test

- Removed streaming and numReturnSequences parameters from ParamCancelReqType.
- Instead, create multiple requests with different parameters in the test.
- Improved output verification by adjusting expected token counts based on request parameters.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
test: Refactor LeaderApiUsageTest

- Removed the streaming parameter from LeaderApiUsageType and adjusted related test logic.
- Renamed the test case from LeaderApiUsageTest to LeaderModeTest to get picked up in CI.
- Updated the test to handle multiple requests with both streaming and non-streaming configurations.
- Adjusted expected outcomes to reflect the new request handling.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #10038](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10038/) [ run ] triggered by Bot

Funatiq committed:
test: Use only InflightFusedBatching in most trtGptModelRealDecoderTest tests

- Modified test instantiations to exclusively use InflightFusedBatching for various test cases, removing InflightBatching.
- This change aims to streamline the testing process and focus on the fused batching approach.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #10038](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10038/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #7408 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7408/) completed with status: 'FAILURE'

Funatiq committed:
fixup! test: Refactor ParamCancelReqTest executor test

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run --stage-list "A10-CPP-1, A30-CPP-1, A30-CPP-2, A30-CPP-3, H100_PCIe-CPP-1, H100_PCIe-CPP-2, DGX_H100-4_GPUs-CPP-1"

tensorrt-cicd commented:
[PR_Github #10124](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10124/) [ run ] triggered by Bot

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #10149](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10149/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7490](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7490/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #10276](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10276/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7594](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7594/) completed with status: 'FAILURE'

Funatiq committed:
test: Increase time to finish in TimeoutTest

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run --only-multi-gpu-test

tensorrt-cicd commented:
[PR_Github #10368](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10368/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7665 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7665/) completed with status: 'SUCCESS'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #10382](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10382/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7676](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7676/) completed with status: 'FAILURE'

Funatiq commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #10391](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10391/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7683](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7683/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #10452](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10452/) [ skip ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10452](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10452/) [ skip ] completed with state `SUCCESS`
Skipping testing for commit `5d52a08`
